rest-api/tests/9-complete-with-all-defence-layers-tests/Tests.System/BaseTests.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.Extensions.Configuration;
+using Xunit.Abstractions;
+
+namespace CompleteWithAllDefenceLayers.Tests.System;
+
+public class BaseTests
+{
+    protected readonly IConfiguration Configuration;
+
+    public BaseTests(ITestOutputHelper output)
+    {
+        var environment = Environment.GetEnvironmentVariable("TEST_ENVIRONMENT") ?? "localhost";
+
+        var builder = new ConfigurationBuilder()
+            .AddJsonFile("Tests.System/testsettings.json", optional: false)
+            .AddJsonFile($"Tests.System/testsettings.{environment}.json", optional: true)
+            .AddEnvironmentVariables();
+
+        Configuration = builder.Build();
+
+        HttpClient = new HttpClient(new ClientCredentialsDelegatingHandler(Configuration));
+        HttpClient.BaseAddress = new Uri(Configuration["BaseAddress:Uri"]!);
+        
+        output.WriteLine("API base uri: {0}", HttpClient.BaseAddress);
+    }
+
+    protected HttpClient HttpClient { get; }
```

Comment:
Rename to something taht is not confused with the class HttpClient, e g TokenHttpClient or HttpClientWithToken?

Reply:
Fixed in amended commit.